lib/puppet/provider/cisco_ospf_area/cisco.rb

```diff
@@ -138,6 +151,48 @@ def properties_set(new_area=false)
           @nu.respond_to?("#{prop}=")
       end
     end
+    # custom setters which require one-shot multi-param setters
+    nssa_set
+  end
+
+  def nssa_set
+    attrs = {}
+    vars = [
+      :nssa,
+      :nssa_route_map,
+      :nssa_default_originate,
+      :nssa_no_redistribution,
+      :nssa_no_summary,
+    ]
+    return unless vars.any? { |p| @property_flush.key?(p) }
+    # At least one var has changed, get all vals from manifest
+    attrs = {}
+    vars.each do |p|
+      if @resource[p] == :default
+        attrs[p] = @nu.send("default_#{p}")
+      else
+        attrs[p] = @resource[p]
+        attrs[p] = PuppetX::Cisco::Utils.bool_sym_to_s(attrs[p])
+      end
+    end
+    hash = {}
+    if attrs[:nssa]
+      hash[:nssa] = attrs[:nssa]
+    else
+      @nu.nssa_set(hash)
+      return
+    end
+    if attrs[:nssa_route_map] == '' || attrs[:nssa_route_map].nil?
+      hash[:route_map] = ''
+      hash[:rm] = ''
+    else
+      hash[:route_map] = 'route-map'
+      hash[:rm] = attrs[:nssa_route_map]
```

Comment:
You shouldn't be setting actual cli syntax in the puppet code. This should be done in NU. I have some related comments for this in the following comment...